### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'entities')

### ISSUE THREAD

**Issue by Jin-xi-bit** [johannesjo/super-productivity#3638]
I have tried many similar applications before, and this one is the best I've used. It has greatly facilitated my work, so kudos to the developer. However, I recently encountered an unsolvable issue on the Linux desktop version and am unsure how to fix it. I hope the developer can help. Nevertheless, I still really like this software.

### Steps to Reproduce

1. Add some modifications to my Linux device and set up WebDAV sync to the server.
2. Set up WebDAV sync on my Android device for downward data synchronization.
3. Add some modifications to my Android device and perform WebDAV synchronization.
4. My Linux device attempts to sync down from WebDAV, but the error appears, preventing synchronization.
5. Then I tried to download `MAIN.json` on my WebDAV server, delete .config/superProductivity and import data from `MAIN.json`.(Because it is similar with the exported data) Then this error appears(details are as below).
![image](https://github.com/user-attachments/assets/d850c76c-9d94-4d7e-a129-79c86e8ae606)
This means "Detected old data while importing. Try to move(?) it? ( I pressed "Ok" here)
![image](https://github.com/user-attachments/assets/daca3138-1df3-4a3c-b293-7ca0eb834606)
This means "Your data seem to be damaged. Try to repair it automaticly? This may cause some data loss. (I pressed "Ok" here)
![image](https://github.com/user-attachments/assets/3d597de1-8b4e-4c7c-b508-61402c486acd)

### Error Log (Desktop only)

[main.log](https://github.com/user-attachments/files/17542410/main.log)

### Console Output

![image](https://github.com/user-attachments/assets/86345081-d68d-4bc6-a984-fe70d5861af2)
![image](https://github.com/user-attachments/assets/5697c681-5739-41f5-8175-05bc64312947)
![image](https://github.com/user-attachments/assets/8eb655a7-e1df-4b7d-985b-af1faa8ae1d5)

### Meta Info
META: SP10.0.11 Electron – zh-CN – Linux x86_64 – Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) superProductivity/10.0.11 Chrome/128.0.6613.162 Electron/32.1.2 Safari/537.36

### Stacktrace
```
taskIdsToRestoreFromArchive.includes (file:///opt/superProductivity/resources/app.asar/dist/browser/src/app/core/data-repair/data-repair.util.ts:230:20)
_fixEntityStates (file:///opt/superProductivity/resources/app.asar/dist/browser/src/app/core/data-repair/data-repair.util.ts:45:16)
_fixEntityStates (file:///opt/superProductivity/resources/app.asar/dist/browser/src/app/core/data-repair/data-repair.util.ts:44:20)
isDataRepairPossible (file:///opt/superProductivity/resources/app.asar/dist/browser/src/app/core/data-repair/data-repair.util.ts:22:12)
constructor (file:///opt/superProductivity/resources/app.asar/dist/browser/src/app/core/data-repair/data-repair.service.ts:17:11)
this._dataRepairService.isRepairPossibleAndConfirmed (file:///opt/superProductivity/resources/app.asar/dist/browser/src/app/imex/sync/data-import.service.ts:117:48)
h_ (file:///opt/superProductivity/resources/app.asar/dist/browser/main-DUZ4F2RX.js:1:1557)
invoke (file:///opt/superProductivity/resources/app.asar/dist/browser/node_modules/zone.js/fesm2015/zone.js:331:157)
```

### Actions Before Error
```
1730118437458: [FocusMode] Set focus session elapsed time 
1730118437566: [SP_ALL] Load(import) all data 
1730118437570: [SP_ALL] All Data was loaded 
1730118441493: [Layout] Toggle SideBar 
1730118446098: [Layout] Hide SideBar (2) 
1730118454380: [Persistence] Save to DB
```

**Comment by johannesjo** [comment]
Thank you very much for reporting! I try to have a look this Friday.

**Comment by jin-xi-arch** [comment]
I just discovered that there are some differences between the `MAIN.json` file uploaded via WebDAV and the `super-productivity-backup.json` file exported using the "Export Data" feature within the app. This is the reason for the error message mentioned earlier, so there’s no need to check the initial error report I submitted. I only need to overwrite the differing content (just some metadata differences) in `MAIN.json`, and I can successfully import data using the "Import from File" feature from `MAIN.json`.

However, **after completing WebDAV synchronization on both ends, once I submit a change from the Android side, an error occurs when the Linux desktop side syncs via WebDAV** (choosing the remote version in case of conflicts). This issue significantly affects desktop usage. I recreated the data and found that this process can consistently trigger the error. Additionally, I attempted to switch from the Linux desktop version to the web version, but a similar error (an unknown error) occurs with a different error page.

This is the error message at the Linux desktop side.
![image](https://github.com/user-attachments/assets/9a3f659c-4d35-49cf-a9b6-4eecf6fc044f)

This is my first time submitting an issue on GitHub. I'm very grateful that, even though I'm just an ordinary user, you're still willing to take the time to address my issue.

**Comment by HalgrimThadeus** [comment]
Could it be linked to the bug i mentioned in the android repo? #https://github.com/johannesjo/super-productivity-android/issues/54 -just a reminder, that maybe helps debugging-

**Comment by jin-xi-arch** [comment]
> Could it be linked to the bug i mentioned in the android repo? #[johannesjo/super-productivity-android#54](https://github.com/johannesjo/super-productivity-android/issues/54) -just a reminder, that maybe helps debugging-

Our data synchronization directions are opposite and my error **occurs at the Linux Desktop** (not Android App). But both occur during WebDAV synchronization, so there might be some connection.

**Comment by johannesjo** [comment]
Unfortunately I am unable to reproduce this myself. Does it help if you delete the remote data once completely manually? 

The error above indicates that the revision id for your archive data saved on android is not correct for some reason (it should never be just a date). 

Possibly related #3617 #3641

**Comment by jin-xi-arch** [comment]
Yes, it helps me. Thank you. There are all steps to fix my problem.
1. Download remote files (`MAIN.json` and `ARCHIVE.json`) and completely **remove them on my WebDAV server**.
2. **Uninstall my android app** to clear any remaining data (which I've never tried before).
3. (Don't do this step unless you also removed data on the Linux device without any backup.) Combine remote files into a complete backup file manually and imported it to the Linux device.
4. **Upload data from the Linux device.**
5. Reinstall app at the Android device and configure WebDAV sync.

**Comment by johannesjo** [comment]
I am glad that it works now. Might be some bug from an earlier version that led to a wrong value for the rev. Glad that it works know and I am sorry for the inconvenience!

**Comment by jin-xi-arch** [comment]
Unfortunately, when I get up at today's morning, this problem occurs again.
![image](https://github.com/user-attachments/assets/1ae839e1-99a3-4186-803f-a383b06abf29)
The last thing I did last night was finish a last day with my Android device (I'm sure I synced it). This morning, I opened my Linux device and tried to sync from remote, and this happened.

There might be some bugs in current version since I just start to use it at version 10.0.11 last week. So it's not because of any earlier version.

**Comment by vincurekf** [comment]
I had this error after unexpected restart. It seems that the `Import from file` function expects to get complete backup file including `taskArchive` which I had in separate file (I had two files `MAIN.json` and ;ARCHIVE.json`). When I merged these files and then imported the merged file the import was successful.
